src/ray/core_worker/transport/out_of_order_actor_submit_queue.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <map>
+#include <utility>
+
+#include "absl/types/optional.h"
+#include "ray/common/id.h"
+#include "ray/core_worker/transport/actor_submit_queue.h"
+
+namespace ray {
+namespace core {
+
+/**
+ * OutofOrderActorSubmitQueue sends request as soon as the dependencies are resolved.
```

Comment:
TODO, we need a description of the guarantee and behaviors of this queue.

Reply:
+ Docstring for each API!